cognite/client/utils/_azure_token_generation.py

```diff
@@ -0,0 +1,40 @@
+import os
+
+from cognite.client.exceptions import CogniteAPIKeyError
+from cognite.client.utils import _auxiliary
+
+
+def _azure_environment_vars_set():
+    azure_api_client_id = os.getenv("AZURE_API_CLIENT_ID")
+    azure_api_client_secret = os.getenv("AZURE_API_CLIENT_SECRET")
+    oidc_authority = os.getenv("OIDC_AUTHORITY")
+
+    return azure_api_client_id is not None and azure_api_client_secret is not None and oidc_authority is not None
+
+
+def _generate_access_token():
+    # Microsoft Authentication Library (MSAL) for Python
+    # https://github.com/AzureAD/microsoft-authentication-library-for-python
+    msal = _auxiliary.local_import("msal")
```

Comment:
We'll likely end up using the ADAL python library (https://github.com/AzureAD/azure-activedirectory-library-for-python) for Aramco (they use ADFS 2016, MSAL only supports ADFS2019), but we don't yet have an ADFS to test with.

Reply:
(if we want to use such an IdP specific library at all. The client credentials flow is pretty simple, IMHO).